### PR TITLE
Line imaging update2

### PR DESCRIPTION
**Pull request by hongliliu**
add the line selection

### CODE REVIEW

**Comment by hongliliu** [comment]
Taking Adam's comments

File: reduction/line_imaging.py [hunk 1]
```diff
@@ -239,7 +243,7 @@
 
                 pars_key = "{0}_{1}_{2}_robust{3}_contsub".format(field, band, arrayname, robust)
                 impars = line_imaging_parameters[pars_key]
-
+                # Todo: should we re-calculate the threshold after the continuum subtraction?
```
Comment (keflavich) [inline reduction/line_imaging.py]: probably yes?

File: reduction/line_imaging.py [hunk 2]
```diff
@@ -195,6 +195,9 @@
                        imagename=lineimagename,
                        restoringbeam='', # do not use restoringbeam='common'
                        # it results in bad edge channels dominating the beam
+                       # I tried this parameter with empty but it resulted in the image not fitting the TP one (in fits file)
+                       # given by the pipeline so that I couldn't combine the two images by feathering. However, if 'common'
+                       # is used, the resulting image can be combined with the TP one in feathering.
```
[the remark this answers is not in the document]
Comment (hongliliu) [reply reduction/line_imaging.py]: Good suggestion. I would try this later.

File: reduction/line_imaging.py [hunk 3]
```diff
@@ -121,15 +119,19 @@
...
             impars = line_imaging_parameters[pars_key]
             # load in the line parameter info
-            linpars = line_parameters[line_name]
+            linpars = line_parameters[field][line_name]
 
             # calculate the channel width
-            count_spws = len(auIMF.spwsforfield(field+'_all_split.ms.contsub', field=field))
-            width = np.max([np.abs(auIMF.effectiveResolutionAtFreq(field+'_all_split.ms.contsub',
+            # todo: replace the file name with each ms set, recalculate the channel width for each
+            # in a 'for' loop, and finally take the maximum
```
[the remark this answers is not in the document]
Comment (hongliliu) [reply reduction/line_imaging.py]: Yes, I tried and it indeed can be done in a metadata tool. Please take it into account later.

**Pull request merged**